src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-form/smb-cluster-form.component.ts

```diff
@@ -0,0 +1,258 @@
+  action: any;
+  resource: string;
+  icons = Icons;
+  domainSettingObject: any;
+  modalData$ = this.smbService.modalData$;
+
+  constructor(
+    private hostService: HostService,
+    private formBuilder: CdFormBuilder,
+    private cd: ChangeDetectorRef,
+    public smbService: SmbService,
+    public actionLabels: ActionLabelsI18n,
+    private orchService: OrchestratorService,
+    private modalService: ModalCdsService,
+    private taskWrapperService: TaskWrapperService,
+    private router: Router
+  ) {
+    super();
+    this.resource = $localize`Cluster`;
+  }
+  ngOnInit() {
+    this.action = this.actionLabels.CREATE;
+    this.smbService.modalData$.subscribe((data) => {
+      this.domainSettingObject = data;
+      this.smbForm.get('domain_setting').setValue(data?.realm);
+    });
+    this.createForm();
+
+    this.action = this.actionLabels.CREATE;
+    this.hostsAndLabels$ = forkJoin({
+      hosts: this.hostService.getAllHosts(),
+      labels: this.hostService.getLabels()
+    }).pipe(
+      map(({ hosts, labels }) => ({
+        hosts: hosts.map((host: any) => ({ content: host['hostname'] })),
+        labels: labels.map((label: string) => ({ content: label }))
+      }))
+    );
+    this.orchStatus$ = this.orchService.status();
+    this.allClustering = ['Default', 'Always', 'Never'];
+  }
+
+  createForm() {
+    this.smbForm = this.formBuilder.group({
+      cluster_id: new FormControl('', {
+        validators: [Validators.required]
+      }),
+      auth_mode: [
+        '',
+        {
+          validators: [Validators.required]
+        }
+      ],
+      intent: [
+        '',
+        {
+          validators: [Validators.required]
+        }
+      ],
+      domain_setting: [''],
+      user_group_setting: [[]],
+      placement: [{}],
+      hosts: [[]],
+      label: [
+        null,
+        [
+          CdValidators.requiredIf({
+            placement: 'label'
+          })
+        ]
+      ],
+      custom_dns: new FormArray([]),
+      jointSources: new FormArray([]),
+      clustering: new UntypedFormControl('Always')
+    });
+
+    this.orchService.status().subscribe((status) => {
+      this.hasOrchestrator = status.available;
+      this.smbForm.get('placement').setValue(this.hasOrchestrator ? 'hosts' : '');
+    });
+  }
+
+  multiSelector(event: any, field: 'label' | 'hosts') {
+    if (field === 'label') this.selectedLabels = event.map((label: any) => label.content);
+    else this.selectedHosts = event.map((host: any) => host.content);
+  }
+
+  onAuthModeChange(): void {
+    const authMode = this.smbForm.get('auth_mode').value;
+    const domainSettingControl = this.smbForm.get('domain_setting');
+    const defineUserPassControl = this.smbForm.get('user_group_setting');
+
+    if (authMode === 'active-directory') {
+      domainSettingControl.setValidators(Validators.required);
+      defineUserPassControl.clearValidators();
+    } else if (authMode === 'user') {
```

Comment:
Ideally, instead of hardcoding values, which is prone to error and harder to reference for modification later on, we can have enums for this kind of scenarios where we know the possible values and will problably be referenced multiple times through the code.
And instead do:
```suggestion
    if (authMode === authMode.ActiveDirectory) {
      domainSettingControl.setValidators(Validators.required);
      defineUserPassControl.clearValidators();
    } else if (authMode === authMode.User) {
```
Or something along the lines